### PR TITLE
feat(compress): support zstd compression

### PR DESCRIPTION
**Pull request by xkx9431**
### What problem does this PR solve?

Issue Number: close/fix/resolve/ref #775 

### What is changed and how it works?

1. The file has been renamed from **gzip.go** to "**handler_compress.go**"
2. Refactored the lazyResponseGzipWriter into an abstract `lazyCompressResponseWriter`
3. Introduced a FixedCachePool

When client query with http header ``"Accept-Encoding: "zstd"), then we change use the zstd compressor to encode the response

### How Has This Been Tested?

In client sdk go side, this works fine with the example
image

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/util/lifted/influx/httpd/handler.go [hunk 1]
```diff
@@ -571,8 +571,9 @@ func (h *Handler) AddRoutes(routes ...Route) {
 		}
 
 		handler = h.responseWriter(handler)
-		if r.Gzipped {
+		if r.CompressSupported {
```
Comment (xkx9431) [inline lib/util/lifted/influx/httpd/handler.go:574]: here, we may need to add a registry for different compressor typeS, I want to plan a refactor item, not in this PR.

File: lib/util/lifted/influx/httpd/zstd.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/klauspost/compress/zstd"
+	"github.com/openGemini/openGemini/lib/cpu"
+)
+
+type lazyZstdResponseWriter struct {
```
Comment (fx408) [inline lib/util/lifted/influx/httpd/zstd.go:13]: 是否可以抽象一个  lazyCompressResponseWriter 对外屏蔽具体的压缩算法

File: lib/util/lifted/influx/httpd/zstd.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	}
+	return w.Writer.Write(p)
+}
+
+type ZstdWriterPool struct {
+	sp    sync.Pool
+	cache chan *zstd.Encoder
+}
+
+func NewZstdWriterPool() *ZstdWriterPool {
```
Comment (fx408) [inline lib/util/lifted/influx/httpd/zstd.go:96]: 建议使用 pool.NewFixedCachePool() 进行包装

**Review by fx408 — APPROVED** [review]

**Review by fx408 — APPROVED** [review]

File: lib/util/lifted/influx/httpd/handler_compress.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
+package httpd
```
Comment (xiangyu5632) [inline lib/util/lifted/influx/httpd/handler_compress.go:1]: The Apache license header is missing. 
Refer：https://github.com/openGemini/openGemini/blob/main/app/ts-monitor/run/handler_metrics.go

Comment (xkx9431) [reply lib/util/lifted/influx/httpd/handler_compress.go:15]: Resloved

**Review by xuthus5 — APPROVED** [review]

**Review by shoothzj — APPROVED** [review]

**Pull request merged**